### PR TITLE
Massive uplift

### PR DESCRIPTION
**Pull request by earsdown**

### CODE REVIEW

**Comment by raphink** [comment]
The day you get that type of PR is the day you regret not writing more specs :wink:

File: manifests/init.pp [hunk 1]
```diff
@@ -1,22 +1,28 @@
-class networkmanager(
-  $version = present,
-  $enable  = true,
-  $start   = true,
-  $gui     = undef,
```
Comment (raphink) [inline manifests/init.pp]: Most of these parameters are lost in the refactoring, meaning a breaking change, which could easily be avoided (at least for backward compatibility's sake).

File: manifests/init.pp [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
+  $openconnect_connections   = {},
+  $openvpn_connections       = {},
+  $wifi_connections          = {},
+) inherits networkmanager::params {
 
-  $openconnect_connections = {},
-  $openvpn_connections     = {},
-  $wifi_connections        = {},
-) {
+  include ::stdlib
```
Comment (raphink) [inline manifests/init.pp]: What's the purpose of including `stdlib` here?

File: manifests/init.pp [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
-  class { '::networkmanager::install': } ->
-  class { '::networkmanager::service': } ->
-  Class['networkmanager']
+  include ::networkmanager::install
+  include ::networkmanager::config
+  include ::networkmanager::service
+
+  Class['networkmanager::install'] ->
+  Class['networkmanager::config'] ~>
+  Class['networkmanager::service']
```
Comment (raphink) [inline manifests/init.pp]: What's the point of splitting this?

File: manifests/install.pp [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
 class networkmanager::install {
-  package { 'network-manager':
-    ensure => $::networkmanager::version,
-  }
-  case $::networkmanager::gui {
-    'gnome': {
-      package { 'network-manager-gnome':
-        ensure => $::networkmanager::version,
-      }
+  include ::networkmanager
```
Comment (raphink) [inline manifests/install.pp]: `networkmanager::install` is a "private" class that's supposed to be called from the `networkmanager` class. There is no point in including the latter here.

File: manifests/install.pp [hunk 5]
```diff
@@ -1,18 +1,23 @@
...
+  if $::networkmanager::manage_packages {
+    package { $::networkmanager::package:
+      ensure => $::networkmanager::package_ensure,
     }
-    'kde': {
-      package { 'plasma-nm':
-        ensure => present,
+    case $::networkmanager::gui {
+      'gnome': {
+        package { $::networkmanager::package_gnome:
```
Comment (raphink) [inline manifests/install.pp]: If that's a clear branch with exclusive branches, there's no point in having two `package` resources. One resource with a fixed title and a namevar should do.

**Comment by raphink** [comment]
After reviewing a bit, I'm not in favor of merging this as is. A lot of this is changing standard coding style into your own preferences.

Please provide a series of separate PRs limited to functional changes, following the standard coding style.

File: manifests/init.pp (hunk 1 above)
Comment (earsdown) [inline manifests/init.pp]: Fair enough. But okay to deprecate?

File: manifests/init.pp (hunk 2 above)
Comment (earsdown) [inline manifests/init.pp]: Two reasons.
1. validate_re() is from stdlib
2. I'd like to use pick() to prefer $version, $enable, $start (vs. $package_ensure, $service_enable, $service_ensure)

**Pull request closed without merging**